byucs340ta/tweeter-samples-java
# M2: Build Tool Updates, Dependency Updates, Bug Fixes, Code Alignment, Test Alignment, etc

**Build Tool Updates**
- Upgrade `gradle` from `4.1.3` to `7.0.3`
- Upgrade `gradle-wrapper` from `6.5` to `7.0.2`
- Add note about `jcenter()` deprecation

**Dependency Updates**
- Remove `androidx.test.ext:junit1:1.1.2` (unused)
- Update `androidx.test.espresso:espresso-core:` from `3.1.1` to `3.4.0`
- Replace `org.junit.jupiter.junit-jupiter.api:5.5.2` with a **downgraded** `junit.junit:4.4.13` so that the style of JUnit (4) matches the mockito style.
- Replace `org.mockito.mockito-all:1.10.19` with `org.mockito.mockito-core:4.0.0`. The former dependency no longer receives updates
- Replace `org.mockito.android:2.20` with `org.mockito.mockito-android:4.0.0`. The former dependency no longer receives updates
- Add `com.squareup.picasso:picasso` for image management
- Target java 11 everywhere

**Bug Fixes**
- AuthToken has a full class

**Library Changes**
- Use Picasso to manage image data and the view

**Code Alignment** (for documents, and future milestones)
- The service layer specifies the type of `Observer` by method (e.g. `LoginObserver`)
- The service layer no longer accepts an observer in the constructor
- The service layer methods take in an additional observer parameter
- BackgroundTask no longer use lambda expressions, instead `loadSuccessBundle()` can be overridden.
- MessageHandler constructors no longer take in a `Looper`, but instead call `super(Looper.getMainLooper());` in their constructor

**Test Alignment**
- `FollowingPresenterTest` is now an example of the same type of tests we are looking for M2C. Specifically, unit tests testing the presenter layer's `handleSuccess`, `handleFailure`, and `handleException` and a test verifying the parameters passed to the service layer are correct.
- Remove service layer test (`FollowingServiceTest`) that does not apply to M2C
- Add a test in both `androidTest/java` and `test/java` to prove that testing dependencies (junit, Mockito spying, Mockito mocks) work right from the start and together instead of just one or the other.
